src/LuaTransformer.ts

```diff
@@ -1229,7 +1238,7 @@ export class LuaTransformer {
                 restParamName,
                 tstl.FunctionExpressionFlags.Declaration
             ),
-            statement
+            statement.pos >= 0 ? statement : classDeclaration // Map to class declaration if constructor was generated
```

Comment:
Why would pos >= 0 indicate it is generated? Couldn't you technically put a constructor at the very end of your class definition?

Reply:
Sorry, I guess this still just wasn't clear. I've reworked it a tad to hopefully be more understandable.